packages/mui-material-next/src/Divider/Divider.tsx

```diff
@@ -20,8 +21,8 @@ const useUtilityClasses = (ownerState) => {
       light && 'light',
       orientation === 'vertical' && 'vertical',
       flexItem && 'flexItem',
-      children && 'withChildren',
-      children && orientation === 'vertical' && 'withChildrenVertical',
+      // TODO: Check next line with maintainers
+      ...(children ? ['withChildren', orientation === 'vertical' && 'withChildrenVertical'] : []),
```

Comment:
We should remove the `withChildrenVertical` composed class, so this line should become only

```suggestion
     children && 'withChildren',
```

This is because we're [dropping composed classes in v6](https://github.com/mui/material-ui/blob/master/packages/mui-material-next/migration.md#remove-composed-css-classes-and-styleoverrides-keys). We should add this to the migration guide, similar to [this example](https://github.com/mui/material-ui/blob/master/packages/mui-material-next/migration.md#removed-combined-anchor-overlap-styleoverrides-keys).

Reply:
Due to the type defined in composeClasses and your request to leave other files untouched, I need to define this line as follows: 
`...(children ? ['withChildren'] : []) `
Is this acceptable?

Reply:
`children && 'withChildren'` should work, right?

Reply:
When the condition is written as `children && 'withChildren'` , considering the "children" types, a situation is created in which "0" is placed in the "root" types, leading to an error. The error message is as follows:

![image](https://github.com/mui/material-ui/assets/21319493/2204ae7d-4d76-4e2f-b11f-1bfb89540bea)

Reply:
Oh, what about `!!children && 'withChildren'` 😊